packages/edit-navigation/src/store/test/resolvers.js

```diff
@@ -189,4 +231,105 @@ describe( 'getNavigationPostForMenu', () => {
 
 		expect( generator.next().done ).toBe( true );
 	} );
+
+	it( 'creates correct core/navigation-link block variations from menu objects', () => {
+		const menuId = 123;
+
+		const generator = getNavigationPostForMenu( menuId );
+
+		// Advance generator
+		generator.next();
+		generator.next();
+		generator.next();
+
+		const menuItems = [
+			{
+				id: 100,
+				title: {
+					raw: 'wp.com',
+					rendered: 'wp.com',
+				},
+				url: 'http://wp.com',
+				menu_order: 1,
+				menus: [ 1 ],
+				parent: 0,
+				classes: [ 'menu', 'classes' ],
+				xfn: [ 'nofollow' ],
+				description: 'description',
+				attr_title: 'link title',
+				object: 'post',
+				object_id: 123, // the post object ID not the menu object ID
+			},
+			{
+				id: 101,
+				title: {
+					raw: 'wp.org',
+					rendered: 'wp.org',
+				},
+				url: 'http://wp.org',
+				menu_order: 2,
+				menus: [ 1 ],
+				parent: 0,
+				classes: [],
+				xfn: [],
+				description: '',
+				attr_title: '',
+				object: 'page',
+				object_id: 456, // the page object ID not the menu object ID
+			},
+			{
+				id: 102,
+				title: {
+					raw: 'wordpress.org',
+					rendered: 'wordpress.org',
+				},
+				url: 'https://wordpress.org',
+				menu_order: 3,
+				menus: [ 1 ],
+				parent: 0,
+				classes: [],
+				xfn: [],
+				description: '',
+				attr_title: '',
+				object: 'custom',
+			},
+		];
+
+		// Feed a known value to the generator yield result
```

Comment:
TBH I still find the [values being passed into `yield`](https://javascript.info/generators#yield-is-a-two-way-street) confusing as it can seem out of step. I've attempted to document the code to explain what is happening.